app/assets/javascripts/visualizations/highvis/baseVis.coffee.erb

```diff
@@ -113,279 +115,258 @@ $ ->
 
-            @delayedUpdate()
+          return
 
-        # Set up accordion
-        globals.configs.yAxisOpen ?= 0
+        # Initialize
+        updateCheckAll = ->
+          numFields = allFields.length
+          numSel = selFields.length
+          indeterminate = numSel > 0 and numSel < numFields
+          $('#ckbx-ckall-' + id).prop('checked', numSel is numFields)
+          $('#ckbx-ckall-' + id).prop('indeterminate', indeterminate)
+
+        # Update the check all box
+        updateCheckAll()
+
+        # Else initialize the checkboxes to the field selection
+        $("input[name='#{id}']").each (key, elem) ->
+          if Number(elem.value) in selFields then $(elem).prop('checked', true)
+
+        # Set handler for the field checkboxes
+        $("input[name='#{id}']").click (e) =>
+          index = Number(e.target.value)
+          if index in selFields then arrayRemove(selFields, index)
+          else selFields.push(index)
+
+          updateCheckAll()
+          handler(index, selFields)
+          @delayedUpdate()
 
-        ($ '#yAxisControl').accordion
-          collapsible:true
-          active:globals.configs.yAxisOpen
+        # Set handler for the check all box
+        $('#ckbx-ckall-' + id).click =>
+          checked = selFields.length isnt allFields.length
 
-        ($ '#yAxisControl > h3').click ->
-          globals.configs.yAxisOpen = (globals.configs.yAxisOpen + 1) % 2
+          # Empty the array without changing the memory address
+          selFields.splice(0, selFields.length)
 
-        ($ '#select-all-y').click =>
-          yFields =
-            if enableDataPoint then data.normalFields
-            else data.normalFields[1..data.normalFields.length]
+          # Repopulate the array if necessary
+          if checked then selFields.push(f) for f in allFields
 
-          if yFields.length != globals.configs.fieldSelection.length
-            ($ '#yAxisControl').find('.y_axis_input').each (i,j) ->
-              ($ j).prop('checked', true)
-            globals.configs.fieldSelection = yFields
-          else
-            for element in data.normalFields
-              arrayRemove(globals.configs.fieldSelection, element)
-            ($ '#yAxisControl').find('.y_axis_input').each (i,j) ->
-              ($ j).prop('checked', false)
+          $("input[name='#{id}']").each (key, elem) ->
+            $(elem).prop('checked', checked)
+
+          updateCheckAll()
+          selected = if checked then allFields else []
+          handler(selected, selFields)
           @delayedUpdate()
 
       ###
-      Draws group selection controls
-      This includes a series of checkboxes and a selector for the grouping field.
-      The checkbox text color should correspond to the graph color.
+      Default handler for radio button yAxis events
+        selected  The new value for the y axis
+        selFields A reference to the object responsible for tracking y selection
```

Comment:
This comment at first didn't make sense to me.  Then I realized `selected` and `selFields` were parameters.  Could you add a `-` between the variables names and their description so its obvious that this isn't a run-off, nonsensical sentence?